pkg/apis/v1beta1/labels.go

```diff
@@ -46,6 +46,7 @@ const (
 	ProviderCompatabilityAnnotationKey = CompatabilityGroup + "/provider"
 	ManagedByAnnotationKey             = Group + "/managed-by"
 	NodePoolHashAnnotationKey          = Group + "/nodepool-hash"
+	NodePoolHashVersionAnnotationKey   = Group + "/nodepool-hash-version"
```

Comment:
Do we want to do a separate annotation or do a version prefix here?

Reply:
The goal will be to use a version annotation, to avoid a string splitting.